### PR TITLE
Improve upload bucket reuse, error handling, and performance

### PR DESCRIPTION
**Pull request by nellh**
This fixes a few issues discovered in the review of #1755 on staging.

1. A new upload bucket was always created for each upload.
2. Performance with many small files suffered.
3. Minor connectivity issues would very rarely hang the upload (1 in 10,000 requests approximately)

### CODE REVIEW

**Review by david-nishi — CHANGES_REQUESTED** [review]
One confusing bit of text is shown when uploading to an existing dataset. "Upload Complete" is shown before the "Begin upload?" prompt.
```
$ node packages/openneuro-cli/src upload --ignoreWarnings '/Users/david/Documents/bids-datasets/6 Mb/ds002293-1.0.0/' -d ds001010
Adding files to "ds001010"
bids-validator@1.5.4

=======================================================================
Upload Complete
To publish the update go to http://localhost:9876/datasets/ds001010 and create a new snapshot
=======================================================================
        Summary:            Available Tasks:        Available Modalities: 
        5 Files, 6MB                                T1w                   
        1 - Subject                                                       
        1 - Session                                                       


        If you have any questions, please post on https://neurostars.org/tags/bids.

=======================================================================
Files to be uploaded:
README - 69 Bytes
? Begin upload? Yes
=======================================================================
Starting a new upload (6a5680c9-30d9-4fff-ad14-713a553efea4) to dataset: 'ds001010'
No files remaining to upload, exiting.
```

## Tested:
- test suite passes
- cli upload
- cli upload existing dataset with no changes; confirmed exit message
- cli upload with datalad container down; confirmed error message

File: packages/openneuro-cli/src/upload.js [hunk 1]
```diff
@@ -156,6 +162,76 @@ export const encodeFilePath = path => {
...
 
+/**
+ * Determine parallelism based on file list
+ * @param {Array<object>} files
+ * @returns {number}
+ */
+export function uploadParallelism(files) {
+  const bytes = uploadSize(files)
+  const averageSize = bytes / files.length
+  const parallelism = averageSize / 524288 // 512KB
```
[the remark this answers is not in the document]
Comment (nellh) [reply packages/openneuro-cli/src/upload.js:173]: This is just a power of two (2^19) that gave approximately right values for test datasets to get good performance when tested on staging. Maybe a smarter thing to do would be to measure it and adjust as the upload runs but this was much better than serial without being too complex to implement.

**Comment by nellh** [comment]
> One confusing bit of text is shown when uploading to an existing dataset. "Upload Complete" is shown before the "Begin upload?" prompt.

This is fixed, stricter TypeScript linting pointed out the error of calling the function instead of defining a new anonymous function.

**Review by david-nishi — APPROVED** [review]
## Tested:
- cli upload output; messages are in order

**Pull request merged**